src/Test/Html/Internal/ElmHtml/ToString.elm

```diff
@@ -115,23 +115,25 @@ nodeRecordToString options { tag, children, facts } =
                     styleValues
                         |> List.map (\( key, value ) -> key ++ ":" ++ value ++ ";")
                         |> String.join ""
-                        |> (\styleString -> "style=\"" ++ styleString ++ "\"")
+                        |> (\styleString -> "style=\"" ++ escapeHtml styleString ++ "\"")
                         |> Just
 
         classes =
             Dict.get "className" facts.stringAttributes
-                |> Maybe.map (\name -> "class=\"" ++ name ++ "\"")
+                |> Maybe.map (\name -> "class=\"" ++ escapeHtml name ++ "\"")
 
         stringAttributes =
             Dict.filter (\k v -> k /= "className") facts.stringAttributes
                 |> Dict.toList
+                |> List.filter (\( k, _ ) -> not (isUnsafeName k))
```

Comment:
React and Preact both filter out invalid attributes. We do the same here.

https://github.com/preactjs/preact-render-to-string/blob/27f340b6e7d77ec7775a49a78d105cad26fa0857/src/index.js#L641
https://github.com/preactjs/preact-render-to-string/blob/27f340b6e7d77ec7775a49a78d105cad26fa0857/src/lib/util.js#L2

https://github.com/facebook/react/blob/ff6283340a10bb72ad0fb16ca027606a9ea1e67c/packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js#L1502
https://github.com/facebook/react/blob/ff6283340a10bb72ad0fb16ca027606a9ea1e67c/packages/react-dom-bindings/src/shared/isAttributeNameSafe.js#L12-L40